LoopKit/InsulinKit/DoseStore.swift

```diff
@@ -954,8 +962,9 @@ extension DoseStore {
     /// - Returns: An array of pump events in the specified order by date
     /// - Throws: An error describing the failure to fetch objects
     private func getPumpEventObjects(matching predicate: NSPredicate, chronological: Bool, limit: Int? = nil) throws -> [PumpEvent] {
+        let recentPredicate = NSPredicate(format: "date >= %@", recentStartDate as NSDate)
         let request: NSFetchRequest<PumpEvent> = PumpEvent.fetchRequest()
-        request.predicate = predicate
+        request.predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [recentPredicate, predicate])
```

Comment:
nit: I think the all the calls to this have existing predicates that are sufficient to not need the `recentPredicate`, and 4 out of 5 of them already filter on date, so I'd recommend removing this, unless there is some other need I'm missing.

Reply:
@ps2 After a thorough analysis, I think you may be right. Would you check me on my thoughts below?

There appears to be 3 different uses of `getPumpEventObjects` that do not directly use date as part of the predicate.

`getNormalizedMutablePumpEventDoseEntries`:
-`NSPredicate(format: "mutable == true && doseType != nil")`
- It manually filters on date range *after* fetching from Core Data, but it doesn't limit what data is fetched from Core Data based upon date.
- Perhaps this isn't too much of an issue because the predicate is only looking for mutable doses (should all be relatively recent, not many of them, and there is already a Core Data index on `mutable`).
- @ps2 Do you think it is safe to rely solely on the `mutable == true` portion of this predicate to not unnecessarily pull in too much data? I think so, but would like your eyes on it.

`getNormalizedPumpEventDoseEntriesForSavingToHealthStore`:
- Uses an OR predicate and the second subpredicate, `NSPredicate(format: "doseType == %@ && mutable == false", DoseType.bolus.rawValue)`, will pull in all immutable boluses over all time.
- This gets called every time a pump manager has new pump events. I think this will be a problem if we don't limit the number of boluses pulled into the result somehow.
- Additionally, this is compounded by the fact that the `syncIdentifier` for every single bolus is then used as a query against the insulin delivery store.
- Ultimately, most of them will be filtered out since they have most likely already been added to the insulin delivery store (and, thus, HealthKit).
- I think removing the `recentPredicate` is also the right choice here, but I'm not sure how to prevent having to pull in and check 90 days worth of boluses every time a pump manager has new pump events.
- @ps2 I'm not sure how to resolve this one. Any ideas?

`lastRecordedPrimeEventDate`:
- `NSPredicate(format: "type = %@", PumpEventType.prime.rawValue)`
- This effectively pulls in every `prime` over all time, but then only uses the very latest (`chronological: false` and `pumpEvents.first`).
- I suppose this one *should* look at all data looking for the latest `prime`.
- Actually, I think the solution to this one is to add the `limit` parameter. (Looking at the sorting functionality of `getPumpEventObjects` it is only relevant to queries for multiple types. Since this only is only looking for events of a single type, `prime`, we can safely ignore the secondary sorting.
- Adding `limit: 1` to the `getPumpEventObjects` call seems like the right approach here. @ps2 What do you think?

Reply:
1) yes, I think `mutable` is fine, and we'd want *all* doses, even if they were old, I think.
2) the first predicate has date in it
3) yes, I think limit works; also, there aren't that many prime events

Reply:
Oh, I see what you're saying about #2. Yeah, this one we can add the recent scope. But I'd add it at the `getNormalizedPumpEventDoseEntriesForSavingToHealthStore` level, and not `getPumpEventObjects`.

Reply:
Okay, thanks. I'll make those changes. :+1:

Reply:
Changes made.